app/src/main/java/io/pslab/sensors/SensorAPDS9960.java

```diff
@@ -209,97 +145,55 @@ public void run() {
-        timeGap = 100;
-        final int step = 1;
-        final int max = 1000;
-        final int min = 100;
-
-        playPauseButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                if (play && scienceLab.isConnected()) {
-                    playPauseButton.setImageResource(R.drawable.circle_play_button);
-                    play = false;
-                } else if (!scienceLab.isConnected()) {
-                    playPauseButton.setImageResource(R.drawable.circle_play_button);
-                    play = false;
-                } else {
-                    playPauseButton.setImageResource(R.drawable.circle_pause_button);
-                    play = true;
-                    if (!indefiniteSamplesCheckBox.isChecked()) {
-                        counter = Integer.parseInt(samplesEditBox.getText().toString());
-                    }
-                }
-            }
-        });
-        sensorDock.setVisibility(View.VISIBLE);
-
-        indefiniteSamplesCheckBox.setChecked(true);
-        samplesEditBox.setEnabled(false);
-        indefiniteSamplesCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
-            @Override
-            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
-                if (isChecked) {
-                    runIndefinitely = true;
-                    samplesEditBox.setEnabled(false);
-                } else {
-                    runIndefinitely = false;
-                    samplesEditBox.setEnabled(true);
-                }
-            }
-        });
-
-        timeGapSeekbar.setMax((max - min) / step);
-        timeGapSeekbar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
-            @Override
-            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
-                timeGap = min + (progress * step);
-                timeGapLabel.setText(timeGap + "ms");
-            }
+    @Override
+    protected void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
 
-            @Override
-            public void onStartTrackingTouch(SeekBar seekBar) {
-                // Do nothing
-            }
+        outState.putInt(KEY_POS_MODE, spinnerMode.getSelectedItemPosition());
 
-            @Override
-            public void onStopTrackingTouch(SeekBar seekBar) {
-                // Do nothing
-            }
-        });
+        outState.putString(KEY_VALUE_RED, tvSensorAPDS9960Red.getText().toString());
+        outState.putString(KEY_VALUE_GREEN, tvSensorAPDS9960Green.getText().toString());
+        outState.putString(KEY_VALUE_BLUE, tvSensorAPDS9960Blue.getText().toString());
+        outState.putString(KEY_VALUE_CLEAR, tvSensorAPDS9960Clear.getText().toString());
+        outState.putString(KEY_ENTRIES_PROXIMITY, tvSensorAPDS9960Proximity.getText().toString());
```

Comment:
Should be `KEY_VALUE_PROXIMITY`

Reply:
Fixed in https://github.com/fossasia/pslab-android/pull/2576/commits/0c00b9b917b90ddf545588f3dd53873a0a11aec1